docs/user_guide/secure_payloads.rst

```diff
@@ -0,0 +1,233 @@
+payload, a key share, called the `u_key` or user key. Only when the Agent has
+passed its enrolment criteria (including any `tpm_policy` or IMA whitelist),
+will the other key share of the decryption key, called the `v_key` or verification
+key, be passed to the Agent by the Keylime Verifier to decrypt the payload.
+
+.. note:: An alternative to secure payloads is to deliver the encrypted data to
+    the node through some other mechanism like `cloud-init` or pre-embedded in a
+    disk image.  The Keylime protocol described above will still run to derive
+    the decryption key for this data, but the data itself will never been seen
+    or transported by Keylime.  This guide does not discuss this method.
+
+Keylime offers two modes for sending secure payloads: single file encryption
+and certificate package mode. In the following sections we describe each.  If
+you're interested in using the more advanced certificate package mode, we
+recommend you also read the Single File Encryption section as it contains
+configuration options and other information that both modes share.
+
+Single File Encryption
+----------------------
+
+In this mode, a file you specify to the `keylime_tenant` application with the
+`-f` option will be encrypted by the Tenant using the bootstrap key and securely
+delivered to the Agent.  Once the Keylime protocol with the Tenant and Verifier
+has completed, the Keylime Agent will decrypt this file and place it in
+`/var/lib/keylime/secure/decrypted_payload` This is the default file name, but
+you can adjust the name of this file using the `dec_payload_file` option in
+`keylime.conf`.  You can also optionally specify a zip file as the file to be
+securely delivered.  If the `extract_payload_zip` option in `keylime.conf` is
+set (which it is by default), then Keylime will automatically extract the zip
+file to `/var/lib/keylime/secure/unzipped`. Finally, Keylime can also execute a
+script contained in the zip file once it has been unzipped.  You can think of
+this like a very simple form of `cloud-init <https://cloudinit.readthedocs.io/>`_.
+By default this script is called `autorun.sh`. You can configure this to some 
+other script with the `payload_script` option in `keylime.conf`.  Note that this 
+script must be contained in the encrypted zip file and must be located in
+`/var/lib/keylime/secure/unzipped`.
+
+Because the keys that Keylime uses to decrypt the data and the decrypted data
+itself are very sensitive, Keylime will only write those files to the
+memory-backed (and therefore non-persistent) `/var/lib/keylime/secure`
+directory. This is a bind-mounted tmpfs partition.  As such, depending on how
+large your payload is, you may need to increase the size of this mounted
+partition by adjusting the `secure_size` option in `keylime.conf`.
+
+This simple mode of operation is suitable for many situations where the secrets
+and other bootstrapping information is basic.  However, there are several
```

Comment:
Minor Nit: I think this should be "are basic"

Reply:
👍